bin/custom/concerttours/src/concerttours/facades/impl/DefaultConcertFacade.java

```diff
@@ -0,0 +1,35 @@
+package concerttours.facades.impl;
+
+import concerttours.data.ConcertData;
+import concerttours.facades.ConcertFacade;
+import concerttours.model.ConcertModel;
+import de.hybris.platform.product.ProductService;
+import de.hybris.platform.servicelayer.dto.converter.Converter;
+import org.springframework.beans.factory.annotation.Required;
+
+public class DefaultConcertFacade implements ConcertFacade {
+    private ProductService productService;
+    private Converter<ConcertModel, ConcertData> concertConverter;
+
+    @Override
+    public ConcertData getConcertDetails(String concertId) {
+        if (concertId == null) {
+            throw new IllegalArgumentException("Concert's id can't be null");
+        }
+        ConcertModel concert = (ConcertModel) productService.getProductForCode(concertId);
+        if (concert == null) {
```

Comment:
Not going to happen. Please follow ProductService.getProductForCode() documentation (can be accessed in IDEA by jumping to this method).